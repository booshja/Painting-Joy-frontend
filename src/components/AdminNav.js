// dependencies
import React from "react";
import styled from "styled-components";
// components
import { NavLink } from "react-router-dom";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledNav = styled.nav`
    display: none;
    background-color: #ffffff;
    min-height: calc(100vh - 60px);
    height: fill;
    width: 300px;
    align-self: stretch;
    border-right: 2px solid #207070;

    ${breakpoints("display", "", [{ 1024: "flex" }])}
    ${breakpoints("flex-direction", "", [{ 1024: "column" }])}
`;

const StyledLink = styled(NavLink)`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
    padding: 1rem 1rem;
    border-bottom: 1px solid #207070;

    &.active {
        font-weight: 700;
        color: #ffffff;
        background-color: #a1c9c9;
        border-left: 4px solid #207070;
    }
`;

const AdminNav = () => {
    return (
        <StyledNav>
            <StyledLink exact to="/admin/dashboard">
                <i className="fas fa-home"></i> Dashboard
            </StyledLink>
            <StyledLink to="/admin/murals">
                <i className="fas fa-paint-roller"></i> Murals
            </StyledLink>
            <StyledLink to="/admin/messages">
                <i className="far fa-envelope"></i> Messages
            </StyledLink>
            <StyledLink to="/admin/items">
                <i className="fas fa-tag"></i> Store Items
            </StyledLink>
            <StyledLink to="/admin/orders">
                <i className="fas fa-truck"></i> Orders
            </StyledLink>
        </StyledNav>
    );
};

export default AdminNav;
