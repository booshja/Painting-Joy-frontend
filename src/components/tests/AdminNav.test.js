import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AdminNav from "../AdminNav";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <AdminNav />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <AdminNav />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Test */
it("renders correctly", () => {
    const { getByText } = render(
        <BrowserRouter>
            <AdminNav />
        </BrowserRouter>
    );

    expect(getByText("Dashboard")).toBeInTheDocument();
    expect(getByText("Murals")).toBeInTheDocument();
    expect(getByText("Messages")).toBeInTheDocument();
    expect(getByText("Store Items")).toBeInTheDocument();
    expect(getByText("Orders")).toBeInTheDocument();
});
