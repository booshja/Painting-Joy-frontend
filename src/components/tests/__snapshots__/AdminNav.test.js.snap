// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: none;
  background-color: #ffffff;
  min-height: calc(100vh - 60px);
  height: fill;
  width: 300px;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  border-right: 2px solid #207070;
}

.c1 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 1rem 1rem;
  border-bottom: 1px solid #207070;
}

.c1.active {
  font-weight: 700;
  color: #ffffff;
  background-color: #a1c9c9;
  border-left: 4px solid #207070;
  cursor: auto;
}

.c1:hover:not(.active) {
  background-color: rgba(161,201,197,0.3);
}

@media screen and (min-width:1024px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<nav
    class="c0"
  >
    <a
      class="c1"
      href="/admin/dashboard"
    >
      <i
        class="fas fa-home"
      />
       Dashboard
    </a>
    <a
      class="c1"
      href="/admin/murals"
    >
      <i
        class="fas fa-paint-roller"
      />
       Murals
    </a>
    <a
      class="c1"
      href="/admin/messages"
    >
      <i
        class="far fa-envelope"
      />
       Messages
    </a>
    <a
      class="c1"
      href="/admin/items"
    >
      <i
        class="fas fa-tag"
      />
       Store Items
    </a>
    <a
      class="c1"
      href="/admin/orders"
    >
      <i
        class="fas fa-truck"
      />
       Orders
    </a>
  </nav>
</DocumentFragment>
`;
